src/conversion/modelconversiondispatcher.js

```diff
@@ -399,10 +404,33 @@ export default class ModelConversionDispatcher {
 			return;
 		}
 
-		const consumable = this._createMarkerConsumable( type, range );
-		const data = { name, range };
+		// In markers case, event name == consumable name.
+		const eventName = type + ':' + name;
 
-		this.fire( type + ':' + name, data, consumable, this.conversionApi );
+		// Fire event for whole marker. This can be used to convert whole marker to element.
+		const markerConsumable = this._createMarkerConsumable( eventName, range );
+		this.fire( eventName, { name, range }, markerConsumable, this.conversionApi );
```

Comment:
Do we fire event on the same range twice from the dispatcher (once for the whole range and once for parts)? It's very strange.